Improve gravity tables presentation on the debug log

rdwebdesign opened the pull request:
### What does this PR aim to accomplish?

1. Add list "type" to the Adlists table. 

    Currently, all lists are shown without differentiation in the log. Only the web interface can differentiate then. If an user mistakenly add all lists as allowlist (antigravity), we can't understand what is wrong without looking at a web interface screenshot or asking for `pihole -g` output.

    <img width="2384" height="116" alt="image" src="https://github.com/user-attachments/assets/fdae79da-c2d0-4c84-b341-8cf24493b6f0" />
    
2. Improve tables readability using text columns


### How does this PR accomplish the above?

1. Adding `type` column to the table:

    <img width="2345" height="120" alt="image" src="https://github.com/user-attachments/assets/b46d60e0-f766-47ba-a68c-c81a81459f08" />

2. Use text columns instead of number codes on gravity tables:
    - Adlists table (`type` column): replace 0 and 1 with `Block` and `Allow`
    - Domains table (`type` column): replace 0, 1, 2 and 3 with `exact-allow`, `exact-deny`, `regex-allow` and `regex-deny`
    - All tables: use `yes`/`no` for all `enabled` columns.

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

rdwebdesign committed:
Add list "type" (block/allow) to the debug log table

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

PromoFaux commented:
I have another idea here - maybe this fits more under request territory - but I have always disliked the 0 1 2 3 4 in the tables - I find it difficult to parse at a glance, even with the key at top of the table.

Much better would be to have something like `CASE type WHEN '0' THEN 'Block' WHEN '1' THEN 'Allow' ELSE type END` Or something like that. 

It would make it easier to scan down the table. Maybe. Thoughts?

yubiuser commented:
Would be easier for sure!

rdwebdesign commented:
I actually tried this on my first attempt, but I didn't like the output. I used "BLOCK" and "ALLOW" (all caps) and the result was confusing to distinguish them. 

I will change to "Allow" and "Block".

rdwebdesign committed:
Show "Block" and "Allow" instead of 0 and 1

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

PromoFaux commented:
Can you please provide a screenshot of this?

rdwebdesign commented:
<img width="1932" height="164" alt="image" src="https://github.com/user-attachments/assets/b2812483-0e29-4ad5-87a1-77b49eb5168a" />

PromoFaux reviewed (dismissed):
Approving, but I've also changed the direction of the PR... if we merge this, we should try to follow suit in the other tables if needed

yubiuser commented:
>  follow suit in the other tables if needed

I think this should happen right away in this PR. Otherwise it will never happen. rd's editor is probably still open ;-)

rdwebdesign commented:
> if we merge this, we should try to follow suit in the other tables if needed

I think we should keep "disabled" and "enabled" as `0` and `1`.

I'm not sure if replacing the 0, 1, 2 and 3 will improve readability.

<img width="864" height="184" alt="image" src="https://github.com/user-attachments/assets/df977c62-54d1-4e3a-9fa0-d765a4b70ab4" />

<img width="888" height="185" alt="image" src="https://github.com/user-attachments/assets/20013747-c7a6-4676-b84f-61c613492187" />

The numbers are on different "columns", making very easy to read and spot the differences between table rows, but their meaning is printed only on the table header.
The texts are more self-explanatory, but the column is larger (11 characters). 

What do you think? Suggestions?

yubiuser commented:
> I'm not sure if replacing the 0, 1, 2 and 3 will improve readability.

In my opinion: they improve it a lot. I really like the type being written, this saves much mental energy.

> The texts are more self-explanatory, but the column is larger (11 characters).

I have many pixel horizontally ;-)

> I think we should keep "disabled" and "enabled" as 0 and 1.

How does it look like with `yes` and `no`. The header says enabled (?) and yes/no would be the natural answer.

rdwebdesign commented:
OK. I agree with your comments on using `type` as text.

About the yes/no:

This is the current format, using 0 and 1:

<img width="800" height="156" alt="image" src="https://github.com/user-attachments/assets/25873559-4e8f-4d85-939e-efd87efbb999" />

This is using yes/no:

<img width="802" height="153" alt="image" src="https://github.com/user-attachments/assets/eced2e63-da92-46de-a59b-88e35c890a1b" />

---

**Note:**

I will use aligned yes/no values.
When I tried using them misaligned (like we had for 0 and 1), the result wasn't very nice and it didn't improve readability: 

<img width="807" height="146" alt="image" src="https://github.com/user-attachments/assets/e5a88376-214b-4741-8f28-d1ce7dea82f2" />

rdwebdesign committed:
Use text colums instead of number codes on gravity tables

- Domains table ("type" column): replace 0, 1, 2 and 3 with
  "exact-allow", "exact-deny", "regex-allow" and "regex-deny"
- All tables: use yes/no for "enabled" columns

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser commented on advanced/Scripts/piholeDebug.sh:
```diff
@@ -1080,15 +1080,15 @@ check_dhcp_servers() {
 
 show_domainlist() {
-    show_db_entries "Domainlist (0/1 = exact allow-/denylist, 2/3 = regex allow-/denylist)" "SELECT id,CASE type WHEN '0' THEN '0   ' WHEN '1' THEN ' 1  ' WHEN '2' THEN '  2 ' WHEN '3' THEN '   3' ELSE type END type,CASE enabled WHEN '0' THEN '   0' WHEN '1' THEN '      1' ELSE enabled END enabled,GROUP_CONCAT(domainlist_by_group.group_id) group_ids,domain,datetime(date_added,'unixepoch','localti ...
+    show_db_entries "Domainlist" "SELECT id,CASE type WHEN '0' THEN 'exact-allow' WHEN '1' THEN 'exact-deny' WHEN '2' THEN 'regex-allow' WHEN '3' THEN 'regex-deny' ELSE type END type,CASE enabled WHEN '0' THEN '  no' WHEN '1' THEN '  yes' ELSE enabled END enabled,GROUP_CONCAT(domainlist_by_group.group_id) group_ids,domain,datetime(date_added,'unixepoch','localtime') date_added,datetime(date_modif ...
```
```suggestion
    show_db_entries "Domainlist" "SELECT id,CASE type WHEN '0' THEN 'exact-allow' WHEN '1' THEN 'exact-deny' WHEN '2' THEN 'regex-allow' WHEN '3' THEN 'regex-deny' ELSE type END type,CASE enabled WHEN '0' THEN '  no' WHEN '1' THEN '  yes' ELSE enabled END enabled,GROUP_CONCAT(domainlist_by_group.group_id) group_ids,domain,datetime(date_added,'unixepoch','localtime') date_added,datetime(date_modified,'unixepoch','localtime') date_modified,comment FROM domainlist LEFT JOIN domainlist_by_group ON domainlist.id = domainlist_by_group.domainlist_id GROUP BY id;" "5 11 7 12 90 19 19 50"
```

I never saw such a long domain in the wild.
(And my screen is not that wide ;-)

rdwebdesign replied:
This is also used for regex entries and I saw some very long examples.

yubiuser replied:
Ok

rdwebdesign replied:
Just an example of a huge regex entry:

<img width="1344" height="233" alt="image" src="https://github.com/user-attachments/assets/2547bd1b-0de7-4e9c-9740-4690232fe963" />

yubiuser replied:
Oh my God! But the is a line break at the end, so the space reserved could easily be a bit shorter.
